framework/framework.go

```diff
@@ -158,21 +161,33 @@ func ProcessCreate(ctx context.Context, obj interface{}, resources []Resource) e
 	}
 
 	for _, r := range resources {
+		if cancelercontext.IsCanceled(ctx) {
+			return nil
+		}
```

Comment:
Here we check if the canceler received a signal. We do this after each step. The code is not that elegant but I couldn't find a way that is really better. We could work with a list of common callbacks and modify variables within the function scope but this would be a bit magic. I like the explicit approach here a bit more. I am open to suggestions though.

Reply:
I like the explicit approach. It's clear and in the framework rather than the operators.

Reply:
We have a concept of resource composition here. Can we us that instead adding those lines?

Reply:
In theory we could have a canceler resource which checks if the context is canceled. But what then? It would need to return an error and we would need to check for this error everywhere. The code lines of code added here would be the same, plus we would need to implement another resource. I also do not like to use error handling for control flow.

Reply:
I think everything would return `nil` immediately since the context is canceled already and this should be the first thing the wrapped resource checks. 

But I just realised  problem is when you wrap resources in different order.